README.md

```diff
@@ -215,3 +215,55 @@ Currently, there are three policy configurations.
 
 A build Task, e.g. one that produces a container image, must abide by both `all-tasks` and
 `build-tasks` policy configurations.
+
+## Task Migration
+
+Task migrations are Bash scripts defined in version-specific task
+directories. In general, a migration consists of a series of `yq` commands that
+modify pipeline in order to work with the new version of task.
+
+Historically, task maintainers write `MIGRATION.md` to notify users what changes
+have to be made to the pipeline. This mechanism is not deprecated. Besides
+writing the file, it is also recommended to write a migration so that the
+updates can be applied to user pipelines automatically.
+
+The following is the steps to write a migration:
+
+- Bump task version. Modify label `app.kubernetes.io/version` in the task YAML file.
+- Ensure `migrations/` directory exists in the version-specific task directory.
+- Create a migration file under the `migrations/` directory. Its name is in
+  form `<new task version>.sh`. Note that the version must match the bumped
+  version.
+
+The migration file is a normal Bash script file. It should
+
+- Accept one argument. The pipeline file path is passed to via this argument.
```

Comment:
We should also mention that this pipeline file is always a tekton Pipeline file. Even if the pipeline is defined inline in a PipelineRun file, the migration tool extracts the Pipeline from it and calls the migration script on the Pipeline.

This also means it's not possible to apply changes that would have to be done on the PipelineRun, e.g. changing the `volumeClaimTemplate` for a workspace or things like that